File: app/src/app/dto/student/student-dto.ts
import { EntityUrl } from "src/app/enum/entity";
import { BaseDto } from "../utilities/base-dto";

export class StudentDto extends BaseDto {
  id: number;
  firstName: string;
  lastName: string;
  age: number;
  adress: string;
  schoolId: number;
  constructor(typeInstance?: string) {
    if (typeInstance) {
      super(typeInstance, EntityUrl.Students);
    } else {
      super('StudentDto', EntityUrl.Students);
    }
    this.id = 0;
    this.firstName = '';
    this.lastName = '';
    this.age = 0;
    this.adress = '';
    this.schoolId = 0;
  }
}

File: api/ManagementProject.Application/Features/Students/StudentDto.cs
using ManagementProject.Application.Features.Dto;

namespace ManagementProject.Application.Features.Students
{
    public record StudentDto : IBaseDto
    {
        public long Id { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public int Age { get; set; }
        public string? Adress { get; set; }
        public long SchoolId { get; set; }
    }
}
